lib/uplink/ext/apikey.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+// #cgo CFLAGS: -g -Wall
+// #include <stdlib.h>
+// #ifndef STORJ_HEADERS
+//   #define STORJ_HEADERS
+//   #include "c/headers/main.h"
+// #endif
+import "C"
+import (
+	"storj.io/storj/lib/uplink"
+)
+
+// ParseAPIKey parses an API Key
+//export ParseAPIKey
+func ParseAPIKey(val *C.char, cErr **C.char) (cAPIKey C.APIKeyRef_t) {
```

Comment:
I just realized that this error handling approach probably isn't too C-like.

```
func ParseAPIKey(val *C.char, cErr **C.char) (cAPIKey C.APIKeyRef_t) {

vs.

func ParseAPIKey(val *C.char, apikey *C.APIKeyRef_t) (err C.APIKeyRefError_t) {
```

Most likely we want the latter, however might be difficult to implement right now.

Reply:
_Something to keep in mind._